File: 53-MaximumSubarray/53-MaximumSubarray.py
# Last updated: 2/21/2026, 9:45:04 AM
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        total=0
        ans=nums[0]
        for num in nums:
            if total<0:
                total=0
            total+=num
            ans = max(ans,total)
        return ans
        

        

File: 53-MaximumSubarray/53-MaximumSubarray.cpp
// Last updated: 2/21/2026, 9:45:06 AM
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int max_g=INT_MIN;
        int max_l=0;
        int n=nums.size();
        int flag=0;
        int m=INT_MIN;
        for(int i=0;i<n;i++)
        {
            if(nums[i]>0)
            {
                flag=1;
                break;
            }
            m=max(m,nums[i]);
    }
        if(flag==0)
        {
            return m;
        }
        for(int i=0;i<n;i++)
        {
            max_l=max_l+nums[i];
            if(max_l>max_g)
            {
                max_g=max_l;
            }
            if(max_l<0)
            {
                max_l=0;
            }
        }
        return max_g;
        
    }
};